app/controller.go

```diff
@@ -771,3 +766,36 @@ func ManageVotes(req *RequestBeginBlock, ctx *context, logger *log.Logger) error
 	}
 	return nil
 }
+
+func addMaturedAmountsToBalance(ctx *context, logger *log.Logger, req *RequestBeginBlock) {
```

Comment:
Is it a good idea to move this func to delegation store? We already have lots of funcs like this in controller.go.

Reply:
this involves both delegation store and balance store, if we move it into delegation store that means we are accessing balance store inside delegation store, is that ok?